build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -169,6 +169,14 @@ function updateReporters(config) {
   if (argv.saucelabs) {
     config.reporters.push('saucelabs');
   }
+
+  const testTypeAllowList = new Set(['unit', 'integration']);
```

Comment:
extreme bike-shedding: "allow list" doesn't seem right here (we allow all types of tests, we generate a JSON report for only some) - so how about calling it something like "JSON_REPORT_TEST_TYPES"?
(also, make this a module-level constant and make it all caps :) )

Reply:
Done, I believe.